### PR TITLE
Apply `black` to `astropy.stats`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to `astropy.stats` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/stats/bayesian_blocks.py [hunk 1]
```diff
@@ -264,8 +264,9 @@ def validate_input(self, t, x=None, sigma=None):
             x += np.zeros_like(t)
 
             if len(unq_t) != len(t):
-                raise ValueError("Repeated values in t not supported when "
-                                 "x is specified")
+                raise ValueError(
+                    "Repeated values in t not supported when " "x is specified"
```
Comment (larrybradley) [inline astropy/stats/bayesian_blocks.py:268]: @WilliamJamieson

File: astropy/stats/bayesian_blocks.py [hunk 2]
```diff
@@ -366,23 +367,23 @@ def fit(self, t, x=None, sigma=None):
 
             # T_k: width/duration of each block
             if 'T_k' in self._fitness_args:
-                kwds['T_k'] = block_length[:R + 1] - block_length[R + 1]
+                kwds['T_k'] = block_length[: R + 1] - block_length[R + 1]
```
Comment (larrybradley) [inline astropy/stats/bayesian_blocks.py:370]: @WilliamJamieson would this be better as `[: (R + 1)]`?

File: astropy/stats/bayesian_blocks.py [hunk 3]
```diff
@@ -485,10 +487,12 @@ def fitness(self, T_k, N_k):
         M_k = T_k / self.dt
         N_over_M = N_k / M_k
 
-        eps = 1E-8
+        eps = 1e-8
         if np.any(N_over_M > 1 + eps):
-            warnings.warn('regular events: N/M > 1.  '
-                          'Is the time step correct?', AstropyUserWarning)
+            warnings.warn(
+                'regular events: N/M > 1.  ' 'Is the time step correct?',
```
Comment (larrybradley) [inline astropy/stats/bayesian_blocks.py:493]: string concat

File: astropy/stats/funcs.py [hunk 4]
```diff
@@ -1244,8 +1278,7 @@ def _kraft_burrows_nousek(N, B, CL):
             return _scipy_kraft_burrows_nousek(N, B, CL)
         except OverflowError:
             if not HAS_MPMATH:
-                raise ValueError('Need mpmath package for input numbers this '
-                                 'large.')
+                raise ValueError('Need mpmath package for input numbers this ' 'large.')
```
Comment (larrybradley) [inline astropy/stats/funcs.py:1281]: string concat

File: astropy/stats/histogram.py [hunk 5]
```diff
@@ -53,8 +58,9 @@ def calculate_bin_edges(a, bins=10, range=None, weights=None):
         # TODO: if weights is specified, we need to modify things.
         #       e.g. we could use point measures fitness for Bayesian blocks
         if weights is not None:
-            raise NotImplementedError("weights are not yet supported "
-                                      "for the enhanced histogram")
+            raise NotImplementedError(
+                "weights are not yet supported " "for the enhanced histogram"
```
Comment (larrybradley) [inline astropy/stats/histogram.py:62]: concat

File: astropy/stats/tests/test_circstats.py [hunk 6]
```diff
@@ -39,57 +39,114 @@ def test_circmean_against_scipy():
...
+                270,
+                119,
+                248,
+                270,
+                45,
+                23,
+            ]
+        )
+        * u.deg
+    )
```
Comment (larrybradley) [inline astropy/stats/tests/test_circstats.py:142]: um, no.  this formatting is much worse.

File: astropy/stats/tests/test_circstats.py [hunk 7]
```diff
@@ -109,15 +166,26 @@ def test_vtest():
...
+            0.5014477,
+            2.6223103,
+            3.7336524,
+            1.8136389,
+            4.1566039,
+            2.7806317,
+            2.4672173,
+            2.8493644,
+        ]
+    )
```
Comment (larrybradley) [inline astropy/stats/tests/test_circstats.py:182]: here too

File: astropy/stats/tests/test_histogram.py [hunk 8]
```diff
@@ -130,8 +129,21 @@ def test_histogram_output_knuth():
...
+            -1.36338114,
+            -0.88255632,
+            -0.4017315,
+            0.07909331,
+            0.55991813,
+            1.04074295,
+            1.52156777,
+            2.00239258,
+        ],
+    )
```
Comment (larrybradley) [inline astropy/stats/tests/test_histogram.py:146]: here too

File: astropy/stats/tests/test_histogram.py [hunk 9]
```diff
@@ -140,19 +152,55 @@ def test_histogram_output():
...
+            -1.02680377,
+            -0.59406143,
+            -0.1613191,
+            0.27142324,
+            0.70416558,
+            1.13690791,
+            1.56965025,
+            2.00239258,
+        ],
+    )
```
Comment (larrybradley) [inline astropy/stats/tests/test_histogram.py:170]: here

File: astropy/stats/tests/test_histogram.py [hunk 10]
```diff
@@ -140,19 +152,55 @@ def test_histogram_output():
...
+        [
+            -2.32503077,
+            -1.59953424,
+            -0.87403771,
+            -0.14854117,
+            0.57695536,
+            1.3024519,
+            2.02794843,
+        ],
+    )
```
Comment (larrybradley) [inline astropy/stats/tests/test_histogram.py:185]: here

File: astropy/stats/tests/test_histogram.py [hunk 11]
```diff
@@ -140,19 +152,55 @@ def test_histogram_output():
...
+            -2.32503077,
+            -1.74087192,
+            -1.15671306,
+            -0.5725542,
+            0.01160465,
+            0.59576351,
+            1.17992237,
+            1.76408122,
+            2.34824008,
+        ],
```
Comment (larrybradley) [inline astropy/stats/tests/test_histogram.py:201]: here

File: astropy/stats/tests/test_jackknife.py [hunk 12]
```diff
@@ -28,10 +28,62 @@ def test_jackknife_stats():
...
+            24,
+            28,
+            26,
+            14,
+            30,
+            28,
+            41,
+            40,
+        ]
+    )
```
Comment (larrybradley) [inline astropy/stats/tests/test_jackknife.py:86]: here

File: astropy/stats/tests/test_sigma_clipping.py [hunk 13]
```diff
@@ -392,64 +381,402 @@ def test_sigma_clip_grow():
...
+            2.17,
+            -0.11,
+            2.0,
+            0.03,
+            0.8,
+            0.12,
+            -0.75,
+            0.58,
+            0.15,
+        ]
```
Comment (larrybradley) [inline astropy/stats/tests/test_sigma_clipping.py:485]: here

File: astropy/stats/tests/test_sigma_clipping.py [hunk 14]
```diff
@@ -392,64 +381,402 @@ def test_sigma_clip_grow():
...
+            84,
+            85,
+            86,
+            90,
+            91,
+            92,
+            93,
+            94,
+        ]
+    )
```
Comment (larrybradley) [inline astropy/stats/tests/test_sigma_clipping.py:521]: here

File: astropy/stats/tests/test_sigma_clipping.py [hunk 15]
```diff
@@ -392,64 +381,402 @@ def test_sigma_clip_grow():
...
+                1,
+                2,
+                0,
+                1,
+                2,
+                0,
+                1,
+                2,
+            ],
+        ]
```
Comment (larrybradley) [inline astropy/stats/tests/test_sigma_clipping.py:650]: here

File: astropy/stats/tests/test_sigma_clipping.py [hunk 16]
```diff
@@ -392,64 +381,402 @@ def test_sigma_clip_grow():
...
+                3,
+                1,
+                2,
+                3,
+                1,
+                2,
+                3,
+                2,
+            ],
+        ]
```
Comment (larrybradley) [inline astropy/stats/tests/test_sigma_clipping.py:763]: here

**Review by larrybradley — COMMENTED** [review]
looks fine except for a few string concatenations and long lists that are broken into one item per line.

File: astropy/stats/bayesian_blocks.py (hunk 1 above)
Comment (WilliamJamieson) [reply astropy/stats/bayesian_blocks.py]: Fixed by 42280b3.

File: astropy/stats/bayesian_blocks.py (hunk 3 above)
Comment (WilliamJamieson) [reply astropy/stats/bayesian_blocks.py]: Fixed by 42280b3.

File: astropy/stats/funcs.py (hunk 4 above)
Comment (WilliamJamieson) [reply astropy/stats/funcs.py]: Fixed by 42280b3.

File: astropy/stats/histogram.py (hunk 5 above)
Comment (WilliamJamieson) [reply astropy/stats/histogram.py]: Fixed by 42280b3.

File: astropy/stats/bayesian_blocks.py (hunk 2 above)
Comment (WilliamJamieson) [reply astropy/stats/bayesian_blocks.py]: See e9adf9e.

File: astropy/stats/tests/test_circstats.py (hunk 6 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_circstats.py]: See e9adf9e.

File: astropy/stats/tests/test_circstats.py (hunk 7 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_circstats.py:137]: See e9adf9e.

File: astropy/stats/tests/test_histogram.py (hunk 8 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_histogram.py]: See e9adf9e.

File: astropy/stats/tests/test_histogram.py (hunk 9 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_histogram.py]: See e9adf9e.

File: astropy/stats/tests/test_histogram.py (hunk 10 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_histogram.py]: See e9adf9e.

File: astropy/stats/tests/test_jackknife.py (hunk 12 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_jackknife.py:40]: See e9adf9e.

File: astropy/stats/tests/test_histogram.py (hunk 11 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_histogram.py]: See e9adf9e.

File: astropy/stats/tests/test_sigma_clipping.py (hunk 13 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_sigma_clipping.py:394]: See e9adf9e.

File: astropy/stats/tests/test_sigma_clipping.py (hunk 15 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_sigma_clipping.py:433]: See e9adf9e.

File: astropy/stats/tests/test_sigma_clipping.py (hunk 14 above)
Comment (WilliamJamieson) [reply astropy/stats/tests/test_sigma_clipping.py:409]: See e9adf9e.

File: astropy/stats/bayesian_blocks.py (hunk 2 above)
Comment (larrybradley) [reply astropy/stats/bayesian_blocks.py]: thanks

File: astropy/stats/tests/test_circstats.py [hunk 17]
```diff
@@ -84,62 +84,23 @@ def test_circmoment():
...
+            85, 324, 340, 157, 238, 254, 146, 232, 122, 329,
+        ]
+    )*u.deg
+
+    beta = np.array(
+        [
+            119, 162, 221, 259, 270, 29, 97, 292, 40, 313, 94, 45,
+            47, 108, 221, 270, 119, 248, 270, 45, 23,
+        ]
+    )*u.deg
```
Comment (larrybradley) [inline astropy/stats/tests/test_circstats.py:101]:
```suggestion
    ) * u.deg
```

File: astropy/stats/tests/test_circstats.py [hunk 18]
```diff
@@ -84,62 +84,23 @@ def test_circmoment():
...
-        * u.deg
-    )
+
+    # fmt: off
+    alpha = np.array(
+        [
+            356, 97, 211, 232, 343, 292, 157, 302, 335, 302, 324,
+            85, 324, 340, 157, 238, 254, 146, 232, 122, 329,
+        ]
+    )*u.deg
```
Comment (larrybradley) [inline astropy/stats/tests/test_circstats.py:94]:
```suggestion
    ) * u.deg
```

**Comment by WilliamJamieson** [comment]
@larrybradley, I have made all the fixes you requested. I also temporarily suppressed the string normalization so that I could run pre-commit and the few issues those processes could solve.

There are a few remaining cleanups that I can see doing:

1. Transition `.format` to f-strings where it has not already been done.
2. Change all the message checks for `pytest.raises` to using the `match` argument.

**Comment by larrybradley** [comment]
Many thanks, @WilliamJamieson.  I wasn't sure if those remaining items are in scope -- are you going to to do those?

**Comment by WilliamJamieson** [comment]
> Many thanks, @WilliamJamieson. I wasn't sure if those remaining items are in scope -- are you going to to do those?

I can if you want me to. I have done enough of these now that I have worked out a pretty efficient process for doing it.

**Comment by larrybradley** [comment]
Yes, please.  Thanks, @WilliamJamieson .

**Comment by WilliamJamieson** [comment]
The f-string and error message cleanups have been done.

**Review by larrybradley — APPROVED** [review]
Many thanks, @WilliamJamieson

**Comment by larrybradley** [comment]
@WilliamJamieson This is ready to merge.

**Pull request merged**